tripy/tripy/backend/mlir/executor.py

```diff
@@ -134,12 +136,8 @@ def execute(self, output_devices: List[device], inputs: List["Tensor"] = []) ->
 
         # Allocate output memory and store buffer pointers.
         outputs = [
-            create_empty_memref(
-                shape=info.shape,
-                dtype=info.dtype,
-                device=info.device,
-                stream=self.stream._active_cuda_stream,
-                use_cache=False,
```

Comment:
@jhalakpatel do we actually need to disable the cache here?

Reply:
Yes, otherwise for two outputs with the same signature, it would use the same buffer. I ran into an issue with one of the tests where both the outputs would end up sharing the same buffer.

Reply:
Isn't that fine for empty memrefs?

Reply:
I think the use of `create_empty_memref` was confusing here since `info.shape` is not meant to be "empty" i.e. as a `0` in of the dim. Instead, it should have been an initialized memref.

I think for "empty" tensors it would be OK to drop the `use_cache`. May be earlier logic was not caching irrespective of whether the shape is empty or not, that is why I ran into the bug.

Reply:
Makes sense. Right now we are only caching for empty tensors, so it should be fine to reuse the same memrefs.